### ISSUE TITLE
sequence errors

### ISSUE THREAD

**Issue by vitalikvoip** [sipwise/rtpengine#763]
I use rtpengine 7.1.1.1  with user-space forwarding and rtpengine reorders some incoming packets introducing out-of-order outgoing rtp packets.
Limiting the number of worker threads to 1 helps to stop reordering. It looks like sometimes an rtpworker receives a packet and then somehow process it too long (context switching?) and by the time the packet it sent out another worker already read, processes and forwarded next packets.
[ooo_stream.pcap.zip](https://github.com/sipwise/rtpengine/files/3131100/ooo_stream.pcap.zip)
1.1.1.1 - sender
2.2.2.2 - rtpengine
3.3.3.3 - receiver

**Comment by rfuchs** [comment]
That's expected to happen with packets that arrive in spurts. Rtpengine makes no efforts to send out packets in sequence.

**Comment by vitalikvoip** [comment]
If there was a mapping of call-id to a single worker it would serialize RTP packets of the same stream. Maybe not a call-id but let's say having a dedicated epoll fd for every poller and distribute calls to all poller threads. Hm, sounds more like a feature request.

**Comment by rfuchs** [comment]
Even then you might get packets reordered by the network stack, or even the network itself. It is UDP after all.

**Comment by vitalikvoip** [comment]
Yes, I absolutely agree reordered packet may come from the network. But when I have a couple of rtpengines in the media path they all reorder packets frequently and some devices can't properly handle this stream probably to a short jitter buffer. They report packet loss in RTCP and video stream sender starts to decrease picture resolution and frame rate.
Anyhow, I running rtpengine with a single poller thread fixes my problem and if I need more throughput I can start multiple rtpengines and load balance calls between them.
